src/ray/core_worker/core_worker.cc

```diff
@@ -127,4 +127,25 @@ std::unique_ptr<worker::ProfileEvent> CoreWorker::CreateProfileEvent(
       new worker::ProfileEvent(profiler_, event_type));
 }
 
+TaskID CoreWorker::GetCallerId() const {
+  TaskID caller_id = GetCurrentTaskId();
```

Comment:
Hmm, I'm a bit worried this may cause a lot of state buildup. Could we use the current worker ID instead which will be bounded in most cases? This will introduce some spurious ordering dependencies but it's probably fine.

Reply:
It seems hacky to me to use the worker ID, and there's no guarantee that that will be bounded. I think for the state buildup, we should just worry about cleaning up the state at the executing actor. We can do this with ref-counting (which I'll work on next).

Reply:
It's a little hacky, but almost strictly better for most cases right?

Reply:
The cases where we would see a difference are not common enough to justify it, I think. You'd need many non-actor tasks that each have a handle to the actor and much fewer nodes than tasks. Therefore, I don't see a point in doing it if it's just going to get fixed with ref-counting anyway.

Reply:
Are we actually going to do ref counting cleanup in the short term though?

This seems like a one-line fix, so why not include it?

Reply:
> Are we actually going to do ref counting cleanup in the short term though?
Yes, I think so.

> This seems like a one-line fix, so why not include it?
It's not actually a one-line fix. We'd also have to change the type for the caller ID (`TaskID` -> `WorkerID`), and we'd have to change the core worker to never clear actor handles. This is really just going to move the state build-up issue into the clients submitting the task instead of the actor executing the task.

Reply:
I see, I didn't realize it was so entangled. I tried doing it myself by changing it to a string but it somehow involves changes to actor registration as well so I gave up.

Btw, can we remove the frontier stuff in actor registration as well?

Reply:
Yeah it requires changes to anything that uses the caller ID as a key, so the scheduling state in the actor registration and the direct actor.

We can remove the frontier stuff, but it's pretty deeply embedded in the raylet codepath. This was originally why I wanted to change the raylet code to schedule based on arrival order, but I decided to just wait until we move completely to direct actors so we can remove all the actor scheduling code from the raylet at once.